chia/wallet/wallet.py

```diff
@@ -174,7 +174,7 @@ def puzzle_for_pk(self, pubkey: G1Element) -> Program:
         return puzzle_for_pk(pubkey)
 
     def puzzle_hash_for_pk(self, pubkey: G1Element) -> bytes32:
-        return puzzle_hash_for_pk(pubkey)
+        return puzzle_for_pk(pubkey).get_tree_hash()
```

Comment:
this looks like a regression. If there is a good reason to do this, I think it should be explained in a comment in the code